functional/test/test_functional.py

```diff
@@ -654,7 +663,7 @@ def test_sum(self):
     def test_average(self):
         l = [1, 2]
         self.assertEqual(1.5, self.seq(l).average())
-        self.assertEquals(4.5, self.seq(l).average(lambda x: x * 3))
```

Comment:
Why would this be deprecated? Even if thats the case that would be independent of the rest of the PR